test/e2e/framework/nodes_util.go

```diff
@@ -331,3 +335,77 @@ func waitForSSHTunnels() {
 		return err == nil, nil
 	})
 }
+
+// NodeNames returns a list of node names.
+func NodeNames(nodes []v1.Node) []string {
+	var names []string
+	for _, node := range nodes {
+		names = append(names, node.Name)
+	}
+	return names
+}
+
+// NodeKiller is a utility to simulate node failures.
+type NodeKiller struct {
+	config   NodeKillerConfig
+	client   clientset.Interface
+	provider string
+}
+
+// NewNodeKiller creates new NodeKiller.
+func NewNodeKiller(config NodeKillerConfig, client clientset.Interface, provider string) *NodeKiller {
+	return &NodeKiller{config, client, provider}
+}
+
+// Run starts NodeKiller until stopCh is closed.
+func (k *NodeKiller) Run(stopCh <-chan struct{}) {
+	for {
+		nodes := k.pickNodes()
+		interval := k.config.MinKillInterval + time.Duration(rand.Float64()*float64(k.config.MaxKillInterval-k.config.MinKillInterval))
+		Logf("Will sleep %v and kill %v nodes at: %v", interval, NodeNames(nodes), time.Now().Add(interval))
```

Comment:
I don't think this log is particularly useful.
I would say that we should log:
- when we kill or restart a given node

The fact that something will happen in few minutes isn't particularly useful, because test may e.g. finish by then.

Reply:
Done.